tests/crypto/test_cases/test_aes_ecb.c

```diff
@@ -0,0 +1,445 @@
+
+	TEST_VECTOR_ASSERT_EQUAL(p_test_vector->expected_err_code, err_code);
+
+	/* Verify the generated AES ciphertext. */
+	TEST_VECTOR_MEMCMP_ASSERT(m_aes_expected_output_buf, m_aes_output_buf,
+				  input_len, p_test_vector->expected_result,
+				  "Incorrect generated AES ciphertext");
+
+	/* Free resources */
+	mbedtls_cipher_free(&ctx);
+}
+
+void monte_carlo_ecb_update_key(size_t key_len, size_t ciphertext_len)
+{
+	if (dbg_hexdump_on) {
+		LOG_HEXDUMP_DBG(m_aes_key_buf, key_len,
+				"m_aes_key_buf before update");
+	}
+	uint8_t divider;
+
+	divider = key_len - ciphertext_len;
+
+	/* Xor previous cipher with key if key_len > cipher_len. */
+	for (uint8_t xor_start = 0; xor_start < divider; xor_start++) {
+		m_aes_key_buf[xor_start] ^=
+			m_prev_aes_output_buf[ciphertext_len - divider +
+					      xor_start];
+	}
+
+	/* Xor cipher with last 16 bytes of key. */
+	for (uint8_t xor_start = 0; xor_start < ciphertext_len; xor_start++) {
+		m_aes_key_buf[divider + xor_start] ^=
+			m_aes_output_buf[xor_start];
+	}
+
+	if (dbg_hexdump_on) {
+		LOG_HEXDUMP_DBG(m_aes_key_buf, key_len,
+				"m_aes_key_buf after update");
+	}
+}
+
+int monte_carlo_ecb(test_vector_aes_t *p_test_vector,
+		    mbedtls_cipher_context_t *p_ctx, size_t key_len,
+		    size_t iv_len, size_t input_len, size_t output_len)
+{
+	uint16_t j;
+	int err_code;
+
+	/* Execution of encryption or decryption 1000 times with same AES key. */
+	for (j = 0; j < 1000; j++) {
+		if (j < 5 && dbg_hexdump_on) {
+			LOG_DBG("MC inner #%d", j);
+			LOG_HEXDUMP_DBG(m_aes_input_buf, input_len,
+					"m_aes_input_buf");
+		}
+		memcpy(m_prev_aes_output_buf, m_aes_output_buf,
+		       sizeof(m_prev_aes_output_buf));
+		err_code =
+			mbedtls_cipher_update(p_ctx, m_aes_input_buf, input_len,
+					      m_aes_output_buf, &output_len);
+		TEST_VECTOR_ASSERT_EQUAL(p_test_vector->expected_err_code,
+					 err_code);
+		memcpy(m_aes_input_buf, m_aes_output_buf, input_len);
+
+		if (j < 5 && dbg_hexdump_on) {
+			LOG_HEXDUMP_DBG(m_aes_output_buf, input_len,
+					"m_aes_output_buf");
+		}
+	}
+
+	/* Update the AES key. */
+	monte_carlo_ecb_update_key(key_len, output_len);
+
+	err_code = cipher_set_key(p_ctx, key_len, p_test_vector->direction);
+	TEST_VECTOR_ASSERT_EQUAL(p_test_vector->expected_err_code, err_code);
+
+	return err_code;
+}
+
+/**@brief Function for the AES Monte Carlo test execution.
+ */
+void exec_test_case_aes_ecb_monte_carlo(void)
+{
+	int err_code;
+
+	TEST_VECTOR_ASSERT_EQUAL(input_len, output_len);
+
+	mbedtls_cipher_context_t ctx;
+	err_code = cipher_init(&ctx, key_len, p_test_vector->mode);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	err_code = cipher_set_key(&ctx, key_len, p_test_vector->direction);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	err_code = cipher_set_padding(&ctx, p_test_vector->padding);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	/* Start the Monte Carlo test. */
+	uint32_t k = 0;
+	start_time_measurement();
+	do {
+		if (k < 3 && dbg_hexdump_on) {
```

Comment:
Another poential influence on measurements

Reply:
`dbg_hexdump_on` is now default false. The user has to set it to true for this to affect runtimes.
The log level would also need to be set to dbg from inf.
A comment is added at the top.

a69739b